.github/workflows/publish.yml

```diff
@@ -2,23 +2,38 @@
 name: Publish
 on:  # yamllint disable-line rule:truthy
   push:
-    branches:
-      - main
+    # branches:
+    #   - main
 jobs:
   ci:
     name: Run ci
     uses: ./.github/workflows/ci.yml
 
+  build:
+    name: Build Alertmanager for all architectures
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        thread: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 ]
+    needs: ci
+    steps:
+      - uses: actions/checkout@v3
+      - uses: prometheus/promci@3cb0c3871f223bd5ce1226995bd52ffb314798b6 # v0.1.0
+      - uses: ./.github/promci/actions/build
+        with:
+          parallelism: 12
+          thread: ${{ matrix.thread }}
   publish_main:
     name: Publish main branch artefacts
     runs-on: ubuntu-latest
-    needs: ci
+    needs: build
     steps:
       - uses: actions/checkout@v3
-      - uses: prometheus/promci@v0.0.2
+      - uses: prometheus/promci@3cb0c3871f223bd5ce1226995bd52ffb314798b6 # v0.1.0
```

Comment:
Why are we downgrading here? That's wrong with the latest version?

Reply:
This is actually an upgrade from `v0.0.2` to `v0.1.0` :) The pinning to a sha I copied from Prometheus, seems like good practice and I intend to do that for the other instances too.

Reply:
🤦 I need to learn to read a bit better - agreed, pinning of versions saves us from unexpected failures.